Luis fix add edit remove descriptions frontend

luisarevalo21 opened the pull request:
# Description
Allow owners to edit a current warning description. 
Only owners are able to edit a warning description. 
Upon editing a warning the remaining warnings become uneditable until save is triggered.

## Related PRS (if any):
This frontend PR is related to the #[983](https://github.com/OneCommunityGlobal/HGNRest/pull/983) backend PR.


## Main changes explained:
- Inside of `WarningTrackerModal.jsx` I changed the p tag to input passing the styles and added the functionality needed.
- The functionality allows owners to edit by changing the same.
- Next a save button was added to trigger the actual save and put action sending the data to the backend.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log in as an owner account
4. Upon loading, click the "+/-" button to load all the current warnings. 
5. Upon loading, create a test, warning and submit it. 
6. Click the newly created warning, and edit it. 
7. Once edited, click the 'save' button to send the updated warning. 
8. View the user's warnings and should display the newly edited warning. Also, the `WarningTrackerModal.jsx`will have the updated list as well. 

## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/fbbfc65f-10ca-492f-8a3c-52b4393cd3e5


## Note:
Include the information the reviewers need to know.
There is a bug, which when creating a new warning, it doesn't allow you to add spaces, I'll fix the issue in this PR #2037

luisarevalo21 committed:
restored tracker modal

luisarevalo21 committed:
WIP: da9c70f33 saving work

luisarevalo21 committed:
added files for frontend add/remove warnings

luisarevalo21 committed:
resolved warning crashing by adding correct method calls for handleToggle

luisarevalo21 committed:
completed frontend

luisarevalo21 committed:
resolved lint issue

TXMO-dev commented on src/actions/warnings.js:
```diff
@@ -61,3 +66,80 @@ export const deleteWarningsById = (warningId, personId) => {
+export const updateWarningDescription = warningDescriptionId => {
+  const url = ENDPOINTS.UPDATE_WARNING_DESCRIPTION(warningDescriptionId);
+
+  return async dispatch => {
```
Great Job here as well but then I have a slight issue because of a potential race condition:
The function dispatch(updateWarningDescriptionAction(res.data)) returns a promise, and its result is then awaited. However, calling dispatch within another dispatch could lead to potential issues such as race conditions or unintended side effects if not handled carefully. Its crucial we look at this for predictable state management. Another issue i also noticed is, The return value of dispatch(updateWarningDescriptionAction(res.data)) is being awaited and then returned. It's important to make sure that this value is what you expect it to be.
Suggested Refactor:
```
export const updateWarningDescription = (warningDescriptionId) => {
  const url = ENDPOINTS.UPDATE_WARNING_DESCRIPTION(warningDescriptionId);

  return async (dispatch) => {
    try {
      // Perform the API call
      const res = await axios.put(url, { warningDescriptionId });
      
      // Dispatch the action and wait for its resolution
      const result = await dispatch(updateWarningDescriptionAction(res.data));

      // Ensure the response payload is returned correctly
      return result.payload;
    } catch (error) {
      // Log the error for debugging purposes
      console.error('Error updating warning description:', error);

      // Return a structured error object
      return { error: error.message || 'An error occurred' };
    }
  };
};
```

TXMO-dev reviewed (changes requested):
Great Job, I left a review kindly take a look at it and let me know your feedback thank you.

JatinAgrawal94 reviewed (changes requested):
New warning can be added. But it's not visible when I try to view it all. I guess it's due to the error visible in the console. Consider this video for more information.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/d438b46a-5e1e-4d93-8beb-19ab5158fcc9

Kavil-Jain-514 reviewed (changes requested):
A new Warning Tracker is not being added to the tracking list.
The video shows the reviewed PR.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/a7084002-817f-42a7-b961-525ccefe61e0



Also, when an old warning is edited. This is the error I encounter
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/e6edaacd-d3af-4bf4-a410-c24a0dca4b48)

luisarevalo21 commented:
> New warning can be added. But it's not visible when I try to view it all. I guess it's due to the error visible in the console. Consider this video for more information.
> 
>  PR.2343.mp4

Hello @JatinAgrawal94 
Please pull latest changes to the backend, I forgot to add backend code to fetch the new warnings. 
As for the warning that appears, you can ignore it as it is irrelevant to this PR as it is error on teamMemberTask, but I didn't touch that file.
Thanks.

luisarevalo21 commented:
> A new Warning Tracker is not being added to the tracking list. The video shows the reviewed PR.
> 
>  PR.2343.mp4 
> Also, when an old warning is edited. This is the error I encounter ![image](https://private-user-images.githubusercontent.com/112303944/338767464-e6edaacd-d3af-4bf4-a410-c24a0dca4b48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzMzk4MzYsIm5iZiI6MTcxODMzOTUzNiwicGF0aCI6Ii8xMTIzMDM5NDQvMzM4NzY3NDY0LWU2ZWRhYWNkLWQzYWYtNGJmNC1hNDEwLWMyNGEwZGNhNGI0OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNFQwNDMyMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02N2NlYjYyODQyMzZjMGIyYmEyNzkzZmJhYjhlNjkxZDg3ZDY5NDJlMTJkN2E3ZTZiNzI5Zjk4YWIxMjc5N2NmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PDNQF0q9gMD802wXF0oh5Wyosv7JO8-ksF5v5jeCjV0)

Hello @Kavil-Jain-514 
I updated the backend as I forgot to add the code to fetch the updated warnings. 
Please pull and run the build command this should resolve the issue of warnings not updating and should fix the post method as well. 
Please re-review it thanks.

akshit0211 commented:
Hi, I am able to add a new warning and edit it. Good job!

<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/f77dc215-987b-4928-884a-545b61cb66f4" alt="warning1">
<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/8236cf04-f0e7-46fe-8ba9-464052cfb86b" alt="added warning 1">
<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/f3c461de-411e-4ccb-a866-07a9aa4fa7f4" alt="warning2">
<img src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112666943/81e9cc2e-f321-4c0c-8d28-f89fe146b6b4" alt="added warning 2">

JatinAgrawal94 reviewed (changes requested):
1) New Warning can be added
2) Existing Warnings can be edited and saved.
3) Other warning becomes disabled while one is being saved after edit.

Issue: When I try to enter the warning name as "NW2" it gives me a message that warning already exists, though it doesn't.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/27bac601-75cb-41e4-b550-e0d4a37dee18

Kavil-Jain-514 reviewed (commented):
I am getting an error when editing a warning.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/9de1792e-54e9-4024-aa42-f708f62afa1a

luisarevalo21 commented:
> I am getting an error when editing a warning.
> 
>  PR.2343.-.983.mp4

Hello 
Try fetching the backend, then do npm run build && npm run start. 
Let me know if it fixes it.

luisarevalo21 committed:
resolved duplicate bug and editing new warning

luisarevalo21 committed:
removed old function

JatinAgrawal94 reviewed (dismissed):
Changes work as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/907c7e08-b581-48a7-829b-9abd97bb0169

Kavil-Jain-514 reviewed (dismissed):
The functionality is working as expected !!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/ef3dd54f-f068-4435-b085-a517e13cc1b8

Vigneshwar-Muriki reviewed (dismissed):
Logged in as the owner and was able to add a new warning, edit it, and received an error message if the warning already exists. Great work!

https://github.com/user-attachments/assets/e4f32ba5-e175-403e-badb-0f4803d0bc0e

![Screenshot 2024-07-13 212356](https://github.com/user-attachments/assets/67a69cae-c8d7-4752-85f4-1a0cf9cae3f5)

kabraambika reviewed (changes requested):
I have verified the PR for both the frontend and related backend. The feature is working, but there's an error appearing in the console whenever there's a warning.

The console shows a 422 Unprocessable Entity error. Please refer to the attached video and screenshot for details.

https://github.com/user-attachments/assets/3600d0ae-63b5-4c67-943b-6baa759aaab0
![image](https://github.com/user-attachments/assets/afbaec2b-22c9-42b9-8623-7f6437f2b967)

SrikanthChowdary495 reviewed (commented):
I am encountering an error. There are API errors.  I am experiencing an error with API. I have attached an image for reference. 
![12](https://github.com/user-attachments/assets/7fa90d2a-5f9a-4096-9a0e-51474b8b5d83)
![11](https://github.com/user-attachments/assets/f488c97b-a03e-433a-b77b-30f00417b42c)

angelalalacheng commented:
Hi,
I encountered the API error as well.

https://www.loom.com/share/3191e0ecddc84d14a19aea9cdfa0c4ac?sid=07371a25-7c6e-4ff5-9fec-0f2111f5c20b

varunreddy03 reviewed (dismissed):
I have checked into both frontend and backend for  this PR and have tested the feature and it is working as expected.

![PR 2343_1](https://github.com/user-attachments/assets/224c593d-d583-4986-af03-8188aa762a65)
![PR 2343_2](https://github.com/user-attachments/assets/e6b72b53-94ff-4ddf-82eb-b8dc6fa1e1a2)
![PR 2343_3](https://github.com/user-attachments/assets/b3d4e8fc-5a94-4141-b743-3d4dbb96079c)

ReinaT5678 commented:
I'm also experiencing an API error:

This pops up when I try to open the "+/-" button under any user:

<img width="1579" alt="Screenshot 2024-08-02 at 7 34 59 PM" src="https://github.com/user-attachments/assets/817e4e87-23e8-4022-8c80-3423352be016">

This is the error when I try to enter a warning tracker title:
<img width="1577" alt="Screenshot 2024-08-02 at 7 33 48 PM" src="https://github.com/user-attachments/assets/d7871757-9f4f-4fbe-aa35-18b2ef99fc99">

luisarevalo21 commented:
> Hello
> Try fetching the backend, then do npm run build && npm run start.
> Let me know if it fixes it.

Hello 

Please check into the backend, then do inside of the folder run the following command `npm run build && npm run start`
Let me know if it fixes it.

luisarevalo21 commented:
> I'm also experiencing an API error:
> 
> This pops up when I try to open the "+/-" button under any user:
> 
> <img alt="Screenshot 2024-08-02 at 7 34 59 PM" width="1579" src="https://private-user-images.githubusercontent.com/56940344/354787553-817e4e87-23e8-4022-8c80-3423352be016.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2NTk0MDUsIm5iZiI6MTcyMjY1OTEwNSwicGF0aCI6Ii81Njk0MDM0NC8zNTQ3ODc1NTMtODE3ZTRlODctMjNlOC00MDIyLThjODAtMzQyMzM1MmJlMDE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDA0MjUwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4MWVmNmIzYWVjNDA5N2ZhZDk2YTk3MzdjMzg2MzU0OTAxYTNmNDJkYWI5YmFmNjNlNDkzZmEwMjA5ODEwNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M6Obt1AWTReMG0FyzH3CbFj6qIruiCBINUm5vi6gkg4">
> This is the error when I try to enter a warning tracker title: <img alt="Screenshot 2024-08-02 at 7 33 48 PM" width="1577" src="https://private-user-images.githubusercontent.com/56940344/354787510-d7871757-9f4f-4fbe-aa35-18b2ef99fc99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2NTk0MDUsIm5iZiI6MTcyMjY1OTEwNSwicGF0aCI6Ii81Njk0MDM0NC8zNTQ3ODc1MTAtZDc4NzE3NTctOWY0Zi00ZmJlLWFhMzUtMThiMmVmOTlmYzk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDA0MjUwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE2MDIwY2QzNGIzZjQ2MmY2ZjQwMTQ1NTA4NTkyZmM0YzRjMTkwOWY4ZGNhMDkxMWFkYTkyYzJjMjNmYzc3NDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DDStRwb4IX6nuU9QAl061dlaz4W2rQjmAEeJPVleWh4">

Hello @ReinaT5678 

Please check into the backend, then do inside of the folder run the following command npm run build && npm run start
Let me know if it fixes it.

luisarevalo21 commented:
> Hi, I encountered the API error as well.
> 
> https://www.loom.com/share/3191e0ecddc84d14a19aea9cdfa0c4ac?sid=07371a25-7c6e-4ff5-9fec-0f2111f5c20b


Hello @angelalalacheng 

Please check into the backend, then do inside of the folder run the following command npm run build && npm run start
Let me know if it fixes it.

luisarevalo21 commented:
Hello @kabraambika 
This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. 
I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? 
Thanks

kabraambika commented:
> Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks

Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with correct implementation then console errors will appear whenever 422 status code will be sent. 
As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.

Changes in backend current code
![image](https://github.com/user-attachments/assets/d7b85ad3-cfa6-49a1-bfef-f60de130a85a)

Changes in frontend current code
![image](https://github.com/user-attachments/assets/9b6bb162-473b-4938-84df-18c275a4c7cd)

Pranay-ai reviewed (dismissed):
Hey @luisarevalo21 
The PR is working as expected here is the screenshot for reference 
![image](https://github.com/user-attachments/assets/84a620bb-e47c-43fe-88a9-2f2313d8169c)

The Modal is having the Latest Updates as well.
Thank YOU

Parth-tech reviewed (dismissed):
Hey Luis,
The PR meets its required functionality and the codebase looks good too. Great job!

Image References:
<img width="1440" alt="Screenshot 2024-08-06 at 5 14 05 PM" src="https://github.com/user-attachments/assets/bb75656e-f654-404f-87a4-e1e6bfb46533">
<img width="1440" alt="Screenshot 2024-08-06 at 5 14 19 PM" src="https://github.com/user-attachments/assets/ff8bc1e4-f920-4826-9e65-02e1b49bdfce">
<img width="1440" alt="Screenshot 2024-08-06 at 5 14 38 PM" src="https://github.com/user-attachments/assets/af7f286b-2c67-449f-b988-832fdb4d68b0">
<img width="296" alt="Screenshot 2024-08-06 at 5 15 00 PM" src="https://github.com/user-attachments/assets/352eee2b-56e6-4e08-9063-3fdb6b8d093c">

arunsrao9 commented:
I am getting this error even after I run the command npm run build && npm run start
<img width="1440" alt="Screenshot 2024-08-06 at 6 13 05 PM" src="https://github.com/user-attachments/assets/13691f01-60c9-4b4e-b785-2c2ac5c4f9a0">
<img width="1440" alt="Screenshot 2024-08-06 at 6 12 59 PM" src="https://github.com/user-attachments/assets/2d157aa3-b7b9-47e3-825f-85ba5d1ea1f5">

luisarevalo21 committed:
resolved merge conflict

luisarevalo21 commented:
> > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> 
> Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> 
> Changes in backend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798214-d7b85ad3-cfa6-49a1-bfef-f60de130a85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTgyMTQtZDdiODVhZDMtY2ZhNi00OWExLWJmZWYtZjYwZGUxMzBhODVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNjI0NTk4ZDE0YzFlZWM2YWMwNmY2YmYxN2JmYTcxOGNmNjliNDMxMDE5OWFkYWIxYjVhMDAzYWE1MThkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1jGKywC0MflupqOYJT6CYZeClzu3vUBiFfB3g3VluIE)
> 
> Changes in frontend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798701-9b6bb162-473b-4938-84df-18c275a4c7cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTg3MDEtOWI2YmIxNjItNDczYi00OTM4LTg0ZGYtMThjMjc1YTRjN2NkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4ZjZiOWNlMmNhNzgxYzVlOThkNmRhMmIxYTE2NGNiM2Y3NjQyMzM1YzkyNjViZDJiMWNjOGIyMTQyMzMxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0-pR3yE7A-18yhELC-SVQwAEgrztmD-vClQynQk2WaY)

Hello @kabraambika 
I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? 
Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.

kabraambika commented:
> > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > 
> > 
> > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > Changes in backend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798214-d7b85ad3-cfa6-49a1-bfef-f60de130a85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTgyMTQtZDdiODVhZDMtY2ZhNi00OWExLWJmZWYtZjYwZGUxMzBhODVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNjI0NTk4ZDE0YzFlZWM2YWMwNmY2YmYxN2JmYTcxOGNmNjliNDMxMDE5OWFkYWIxYjVhMDAzYWE1MThkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1jGKywC0MflupqOYJT6CYZeClzu3vUBiFfB3g3VluIE)
> > Changes in frontend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798701-9b6bb162-473b-4938-84df-18c275a4c7cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTg3MDEtOWI2YmIxNjItNDczYi00OTM4LTg0ZGYtMThjMjc1YTRjN2NkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4ZjZiOWNlMmNhNzgxYzVlOThkNmRhMmIxYTE2NGNiM2Y3NjQyMzM1YzkyNjViZDJiMWNjOGIyMTQyMzMxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0-pR3yE7A-18yhELC-SVQwAEgrztmD-vClQynQk2WaY)
> 
> Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.

Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.

AnkitLall reviewed (changes requested):
I have tested the feature, following the instructions in the PR, and the update feature works and duplicate entry error works too. Please refer the screen recording attached below.

Although, I found a small discrepancy. The warning text in the modal gets changed to Title Case while the text in the warnings list remains unchanged. Refer the attached screenshot.


https://github.com/user-attachments/assets/ee13b6d5-9529-4466-a3ba-f94c49a7fdff

<img width="1512" alt="Screen Shot 2024-08-10 at 2 57 58 PM" src="https://github.com/user-attachments/assets/aedac010-f624-4dd0-b70c-7310c89b76c8">

adityasure9 reviewed (dismissed):
The functionality to add, edit, and remove warnings works as expected. The ability for owners to manage warnings has been implemented well, and the updates reflect correctly in the UI. 

![1](https://github.com/user-attachments/assets/335b3667-3e55-4477-b965-cc403f4926ea)

luisarevalo21 commented:
> I have tested the feature, following the instructions in the PR, and the update feature works and duplicate entry error works too. Please refer the screen recording attached below.
> 
> Although, I found a small discrepancy. The warning text in the modal gets changed to Title Case while the text in the warnings list remains unchanged. Refer the attached screenshot.
> 
>  Screen.Recording.2024-08-10.at.2.29.09.PM.mp4 
> <img alt="Screen Shot 2024-08-10 at 2 57 58 PM" width="1512" src="https://private-user-images.githubusercontent.com/26919475/356833322-aedac010-f624-4dd0-b70c-7310c89b76c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1MjA4MzEsIm5iZiI6MTcyMzUyMDUzMSwicGF0aCI6Ii8yNjkxOTQ3NS8zNTY4MzMzMjItYWVkYWMwMTAtZjYyNC00ZGQwLWI3MGMtNzMxMGM4OWI3NmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDAzNDIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxODhlNTk2NWFmYjg1NzIzNTJkNzhjYzA4ZDViYTM1YWY3MzJkYTc1NWY5ZmI0MDExN2NjZGFlN2NlYzllYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Lzwo3sVOUBoZtWnszXR7qEUCTUhIIfqRXzwc5AKCMK8">

Hello @AnkitLall 
That was the intended functionality, was to store, the warnings as lowercased and CSS was used to uppercase the warning. 
However, I can see the argument as whatever the owner types should be stored. My issue would be the CSS will capitalize on the modal I guess I can remove it. What are your thoughts? I'd like to hear them.

luisarevalo21 commented:
> > > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > > 
> > > 
> > > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > > Changes in backend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798214-d7b85ad3-cfa6-49a1-bfef-f60de130a85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTgyMTQtZDdiODVhZDMtY2ZhNi00OWExLWJmZWYtZjYwZGUxMzBhODVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNjI0NTk4ZDE0YzFlZWM2YWMwNmY2YmYxN2JmYTcxOGNmNjliNDMxMDE5OWFkYWIxYjVhMDAzYWE1MThkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1jGKywC0MflupqOYJT6CYZeClzu3vUBiFfB3g3VluIE)
> > > Changes in frontend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798701-9b6bb162-473b-4938-84df-18c275a4c7cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTg3MDEtOWI2YmIxNjItNDczYi00OTM4LTg0ZGYtMThjMjc1YTRjN2NkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4ZjZiOWNlMmNhNzgxYzVlOThkNmRhMmIxYTE2NGNiM2Y3NjQyMzM1YzkyNjViZDJiMWNjOGIyMTQyMzMxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0-pR3yE7A-18yhELC-SVQwAEgrztmD-vClQynQk2WaY)
> > 
> > 
> > Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.
> 
> Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.

Hello @kabraambika 
Thanks for the response, I'll go ahead and use a 200, it will remove the 422 on the console. I'll ping you once I push the changes. 
Thanks!

luisarevalo21 commented:
> > > > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > > > 
> > > > 
> > > > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > > > Changes in backend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798214-d7b85ad3-cfa6-49a1-bfef-f60de130a85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTgyMTQtZDdiODVhZDMtY2ZhNi00OWExLWJmZWYtZjYwZGUxMzBhODVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNjI0NTk4ZDE0YzFlZWM2YWMwNmY2YmYxN2JmYTcxOGNmNjliNDMxMDE5OWFkYWIxYjVhMDAzYWE1MThkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1jGKywC0MflupqOYJT6CYZeClzu3vUBiFfB3g3VluIE)
> > > > Changes in frontend current code ![image](https://private-user-images.githubusercontent.com/90600253/354798701-9b6bb162-473b-4938-84df-18c275a4c7cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwODY3ODQsIm5iZiI6MTcyMzA4NjQ4NCwicGF0aCI6Ii85MDYwMDI1My8zNTQ3OTg3MDEtOWI2YmIxNjItNDczYi00OTM4LTg0ZGYtMThjMjc1YTRjN2NkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA4VDAzMDgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4ZjZiOWNlMmNhNzgxYzVlOThkNmRhMmIxYTE2NGNiM2Y3NjQyMzM1YzkyNjViZDJiMWNjOGIyMTQyMzMxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0-pR3yE7A-18yhELC-SVQwAEgrztmD-vClQynQk2WaY)
> > > 
> > > 
> > > Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.
> > 
> > 
> > Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.
> 
> Hello @kabraambika Thanks for the response, I'll go ahead and use a 200, it will remove the 422 on the console. I'll ping you once I push the changes. Thanks!

Hello @kabraambika 
I adjusted the 422 error and no longer returns a post method on the frontend, and changed it to a 200.

SMitta11 reviewed (dismissed):
I login as owner and can see +-below users name after running frontend and backend:
![Screen Shot 2024-08-17 at 5 07 18 PM](https://github.com/user-attachments/assets/1ff382c5-1c4a-4a76-a078-6828fe919403)
Added a new warning
![Screen Shot 2024-08-17 at 5 07 57 PM](https://github.com/user-attachments/assets/72763a6e-7555-4a2c-8f77-3d7e6c756fb0)
Then clicked again on tracking and can see newly added warning:
![Screen Shot 2024-08-17 at 5 08 38 PM](https://github.com/user-attachments/assets/57bde61a-7fe6-4774-98d6-faea56719ec7)
As given in PR, edited this new warning and saved it:
![Screen Shot 2024-08-17 at 5 11 43 PM](https://github.com/user-attachments/assets/5e3d4994-a82d-49c3-87f7-0da39ef13f5c)
Then clicked on tracking and can see this new warning
![Screen Shot 2024-08-17 at 5 12 15 PM](https://github.com/user-attachments/assets/dced2bdb-ba7d-4f92-9695-3426128d7c00)
Checked that this warning is appearing in front of other user's name:
![Screen Shot 2024-08-17 at 5 12 26 PM](https://github.com/user-attachments/assets/72274b31-be91-4ce6-aaf3-a81b45b1f88c)
Also checked the console:
![Screen Shot 2024-08-17 at 5 16 05 PM](https://github.com/user-attachments/assets/359afdab-a652-4eb3-8838-5c4c094416c8)

luisarevalo21 committed:
adjusted styling for new warning form

Lemonnycodes reviewed (dismissed):
Hi,

I tested the PR, including creating, editing, and displaying warnings. Everything works as expected.

Reference Images:
![PR 2343 (1)](https://github.com/user-attachments/assets/7ef34dfd-276f-42b1-a338-5ac28ba9122e)
![PR 2343 (2)](https://github.com/user-attachments/assets/019f3c5e-8a96-499f-8a3c-58f734a79c8c)
![PR 2343 (3)](https://github.com/user-attachments/assets/6345c2e7-495c-4afc-a109-397be5b9c1c5)
![PR 2343 (4)](https://github.com/user-attachments/assets/f81b8769-bb68-4fa9-9f02-66c8daaad269)
![PR 2343 (5)](https://github.com/user-attachments/assets/3151f81e-0cc0-4446-ae52-a655ed17a5d0)

yashwanth170 reviewed (dismissed):
The functionality works as expected. Attaching screenshots.
![test_demo_1](https://github.com/user-attachments/assets/fda46eff-e377-4cc6-bca3-bfd831b85a80)
![test_demo_2](https://github.com/user-attachments/assets/24214061-668e-4a70-b112-08ee66b76abb)
![test_demo_3](https://github.com/user-attachments/assets/b984f162-40e2-4938-9927-20d7a325bf97)

AJAYINAVOLU reviewed (dismissed):
Changes are working as expected. 

<img width="1420" alt="PR 2343" src="https://github.com/user-attachments/assets/9a8f8696-e6ad-4631-aac8-7befea4db383">
<img width="1411" alt="PR 2343-1" src="https://github.com/user-attachments/assets/c44f7d65-1504-4e86-aa86-da03bf62755e">

Thank You

ishangoel17 reviewed (commented):
Verified the PR and the changes are working as expected. Able to add new warning and view/edit it. However, on editing and saving the edited warning, a space appears on the top of the warnings list as shown in screenshot5. Sometimes for a short second there is message saying the warning edit has been saved successfully but its intermittent and does not appear every time on editing a warning. On the times, the message does not appear, there is just the empty space as shown in screenshot5. Below are the screenshots for reference:
![Capture2343](https://github.com/user-attachments/assets/ff7b177b-ff66-4bc9-8f30-74b8ff9197b6)
![Capture2343_2](https://github.com/user-attachments/assets/39a9ae8e-8055-489d-b86d-c752a8809fdb)
![Capture2343_3](https://github.com/user-attachments/assets/0f4765d5-e193-4e04-86b0-3c9a9d51d9dc)
![Capture2343_4](https://github.com/user-attachments/assets/9a7e4b68-00e9-43ae-a65d-6ecd73be7b9d)
![Capture2343_5](https://github.com/user-attachments/assets/b40de8cb-0345-4802-8173-072395d433eb)

Shreyav2000 reviewed (changes requested):
The warnings are being added and deleted properly as intended, but the system is allowing duplicate warnings to be created, which it shouldn’t. This issue arises despite the presence of the checkForDuplicates function, which is designed to prevent this behavior. The problem likely stems from the function not being executed at the correct point in the process or not properly comparing warning titles. It could be failing due to inconsistencies like case sensitivity, spacing, or trimming that prevent it from detecting duplicates effectively. I am attaching a video below showing the same and a screenshot for more clarification. Thank you!

https://github.com/user-attachments/assets/7d4ec9f9-2eda-4854-83a9-d4ccf6f008b5

<img width="485" alt="Screenshot 2024-09-06 at 12 40 35 AM" src="https://github.com/user-attachments/assets/f5750b9f-8564-4648-b318-e23ac39db8fd">

vedantg24 reviewed (dismissed):
Tested the PR, and it is working as expected. Warnings can be added and edited successfully. Also, tried to add a duplicate warning, and it displayed the error for “warning already exists”.

![Screenshot 2024-09-06](https://github.com/user-attachments/assets/e0e6d227-fdf5-4352-9a09-ae0430a32009)
![Screenshot 2024-09-06](https://github.com/user-attachments/assets/71c46268-d036-4fed-97cb-deef81f34271)
![Screenshot 2024-09-06](https://github.com/user-attachments/assets/f7010142-f3a7-4cdb-bc8e-9b6d773939b6)
![Screenshot 2024-09-06](https://github.com/user-attachments/assets/a60f9f33-7631-491e-a646-ce8475b1a602)

luisarevalo21 commented:
> The warnings are being added and deleted properly as intended, but the system is allowing duplicate warnings to be created, which it shouldn’t. This issue arises despite the presence of the checkForDuplicates function, which is designed to prevent this behavior. The problem likely stems from the function not being executed at the correct point in the process or not properly comparing warning titles. It could be failing due to inconsistencies like case sensitivity, spacing, or trimming that prevent it from detecting duplicates effectively. I am attaching a video below showing the same and a screenshot for more clarification. Thank you!
> 
>  Screen.Recording.2024-09-06.at.12.41.12.AM.mov 
> <img alt="Screenshot 2024-09-06 at 12 40 35 AM" width="485" src="https://private-user-images.githubusercontent.com/114371561/365050552-f5750b9f-8564-4648-b318-e23ac39db8fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwMjY0MDIsIm5iZiI6MTcyNjAyNjEwMiwicGF0aCI6Ii8xMTQzNzE1NjEvMzY1MDUwNTUyLWY1NzUwYjlmLTg1NjQtNDY0OC1iMzE4LWUyM2FjMzlkYjhmZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQwMzQxNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjgzMjE1MmU1NDY2Y2YyZGY0OTA0NThlOGFmY2E1MzI4Y2I5NDRlYjBmM2M3MzhhNTlhYWJlNGUyMjg3OTEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.liH67AGkw32EIfxEE63SzgytUgBgKIl6WWhLiBR1JwA">

Hello @Shreyav2000 
Please pull the most recent code from this branch. 
Run the the command 'git pull' while on this branch and should fetch the most updated code. 
I tried to duplicate your error, but shows I cannot have duplicates. 
Also, when running the backend pull as well, and run the command 'npm run build && npm run start' 
I've attached a video of my example. 
Thanks! 

https://github.com/user-attachments/assets/950496a9-fe20-4e9c-973e-2525ec8d9ffd

luisarevalo21 committed:
resolved merge conflicts

luisarevalo21 committed:
restored tracker modal

saurabhshetty29 commented:
Encountered the following error when clicking on the '+/-' button:
<img width="1064" alt="Screenshot 2024-09-10 at 11 31 16 PM" src="https://github.com/user-attachments/assets/ee15ed99-5ae8-473b-a817-5f67809a70ac">

An error occurs when attempting to add a warning:
<img width="486" alt="Screenshot 2024-09-10 at 11 31 44 PM" src="https://github.com/user-attachments/assets/e500a01d-c707-4d66-ac40-908c614bf5de">

luisarevalo21 committed:
added hover effect for permanent warnings

luisarevalo21 commented:
> Encountered the following error when clicking on the '+/-' button: <img alt="Screenshot 2024-09-10 at 11 31 16 PM" width="1064" src="https://private-user-images.githubusercontent.com/111922055/366326437-ee15ed99-5ae8-473b-a817-5f67809a70ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTMwMjcsIm5iZiI6MTcyNjExMjcyNywicGF0aCI6Ii8xMTE5MjIwNTUvMzY2MzI2NDM3LWVlMTVlZDk5LTVhZTgtNDczYi1hODE3LTVmNjc4MDlhNzBhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMlQwMzQ1MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOGJiYzJmMDhkNjQwMDA3NDI4ZWJmZGY2ODM3OThiOTJhMzE1OGFmYzJiMDk2ZDA1YjE0ZDQ5YmYxNzdlODljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HNyqH6NMLx7v59fpGCNwGOmvyvwWEKTTrcVFSKDWyQY">
> 
> An error occurs when attempting to add a warning: <img alt="Screenshot 2024-09-10 at 11 31 44 PM" width="486" src="https://private-user-images.githubusercontent.com/111922055/366326562-e500a01d-c707-4d66-ac40-908c614bf5de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTMwMjcsIm5iZiI6MTcyNjExMjcyNywicGF0aCI6Ii8xMTE5MjIwNTUvMzY2MzI2NTYyLWU1MDBhMDFkLWM3MDctNGQ2Ni1hYzQwLTkwOGM2MTRiZjVkZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMlQwMzQ1MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MTI2ZjA1OGJkODA0NWFlOGZhNTg0OGRiMTUzYWU3NWQyZmY0YjYwYTEyYjU5ODczYzdlNGRmNmNiYWUwNTBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XnTJt_IJtv5FCjGNrWgSVlh4Y2-esxlQYBxjsTjHkr0">

@saurabhshetty29 please ensure you've pulled the latest changes. 
also make sure you pulled the appropriate backend as well as stated in the post. 
If you still have issues feel free to reach out.

crystallow1168 reviewed (dismissed):
Hi @luisarevalo21,

Changes looks good and working as expected. Below are my findings:

https://github.com/user-attachments/assets/633fb9bd-7f5c-4497-94f2-81fdb0b5ce2d


✅ Owner user can create a warning and warning can be seen when clicked on the "Tracking" button.
✅ Owner user can edit the warning and changes can be seen under the tracking warning section

sud306 commented:
Hello,
I reviewed PR983 and PR2343 and found that it's working as expected.

![9 PR983_PR2343_1](https://github.com/user-attachments/assets/aa89cb2d-f925-4ab0-a591-7549cd7c719a)

![9 PR983_PR2343_2](https://github.com/user-attachments/assets/76176f13-0f8e-4cb3-85f3-29b894755052)

SFA23SCM35V reviewed (dismissed):
I have reviewed both frontend and backend PR's and no defect found 
Great job Approving the change 
![1](https://github.com/user-attachments/assets/2fac575d-1ab7-4be8-9bda-785719e2e799)
![2](https://github.com/user-attachments/assets/bcbf57ed-791a-4676-b6c5-e9dab8921b11)
![3](https://github.com/user-attachments/assets/93892590-bc5c-4d4d-bbc3-76eab0243542)
![4](https://github.com/user-attachments/assets/bb337f69-64bf-4879-9a5b-bfe780bfd404)
![5](https://github.com/user-attachments/assets/9b7b0c86-88dd-4143-af0b-df1e003ac02b)
![6](https://github.com/user-attachments/assets/1cc3f9f5-96b9-47e1-b39b-072aaa5f6d74)
![Uploading 7.png…]()

neeharikakoniki reviewed (approved):
The warnings creation and editing functionality works smoothly. The UI updates immediately, and the WarningTrackerModal.jsx reflects the changes accurately.

![image](https://github.com/user-attachments/assets/d7733753-ef8a-4362-923b-07d4464569a0)
![image](https://github.com/user-attachments/assets/49e9339d-6494-49e6-93c7-7412881ffad9)

ashayk9 reviewed (commented):
I used the proper frontend and backend branches as per the instructions.
But encountered the following error when clicking on the '+/-' button:
![Screenshot 2024-09-19 at 12 31 52 PM](https://github.com/user-attachments/assets/184cebba-257b-4157-80cd-90e709c9bf79)
![Screenshot 2024-09-19 at 12 32 54 PM](https://github.com/user-attachments/assets/9d771e45-cb4e-4ca9-9ed7-9f52604aebcb)
Let me know if I need to test it in a different way

Muzammil603 reviewed (changes requested):
![image](https://github.com/user-attachments/assets/1ffa64d9-ba44-467e-9223-97441e98ba13)
I am not getting +/- option

pallavithorat commented:
I tested the PR and encountered an issue where the "Add New Warning Tracker" functionality is returning an error message, "error occurred" (screenshot attached).
<img width="1470" alt="ss" src="https://github.com/user-attachments/assets/e44b8c79-790a-4cbf-b22b-2416185a68ba">
<img width="1470" alt="Screenshot 2024-10-06 at 8 18 37 PM" src="https://github.com/user-attachments/assets/43b2b85f-edf2-4c4d-9f38-d6edd89f71bb">

Please let me know if there's a different way I should be testing this.

luisarevalo21 commented:
> I tested the PR and encountered an issue where the "Add New Warning Tracker" functionality is returning an error message, "error occurred" (screenshot attached). <img alt="ss" width="1470" src="https://private-user-images.githubusercontent.com/87472620/374014200-e44b8c79-790a-4cbf-b22b-2416185a68ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NDI0NzUsIm5iZiI6MTcyODQ0MjE3NSwicGF0aCI6Ii84NzQ3MjYyMC8zNzQwMTQyMDAtZTQ0YjhjNzktNzkwYS00Y2JmLWIyMmItMjQxNjE4NWE2OGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA5VDAyNDkzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyMjQzZDhjZjMxZjlkY2UzN2Y3MTFlNzU3NDA4ZmU4NGU4N2Y3ZGZlMGY3YTJkMmY3OGU0YTBiYTJkMmExOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OnBJyBGLDt7zme3oVEI3Vyo5WKuLdEs6bJMJOyds5fc"> <img alt="Screenshot 2024-10-06 at 8 18 37 PM" width="1470" src="https://private-user-images.githubusercontent.com/87472620/374014225-43b2b85f-edf2-4c4d-9f38-d6edd89f71bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NDI0NzUsIm5iZiI6MTcyODQ0MjE3NSwicGF0aCI6Ii84NzQ3MjYyMC8zNzQwMTQyMjUtNDNiMmI4NWYtZWRmMi00YzRkLTlmMzgtZDZlZGQ4OWY3MWJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA5VDAyNDkzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyNTY3ZjU1Y2Q3NWZmMzAyNmRkYWI4ZWVjNGExYzE4ZTI3YThhOTQxY2NlNGRiMTQ4MjNiM2Y2ZWI4ZjM0MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1zczbcOU6vednE57fdQ2s5U4ARu7DL63c5oauZgDOyU">
> 
> Please let me know if there's a different way I should be testing this.

Hello 
Please ensure the frontend is running node 14.17.6. 
Also, follow the backend and run npm run build && npm run start to run and build the backend server. 
Reach out to me if you need further assistance.

luisarevalo21 commented:
> ![image](https://private-user-images.githubusercontent.com/145288870/369663737-1ffa64d9-ba44-467e-9223-97441e98ba13.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NDI0NzYsIm5iZiI6MTcyODQ0MjE3NiwicGF0aCI6Ii8xNDUyODg4NzAvMzY5NjYzNzM3LTFmZmE2NGQ5LWJhNDQtNDY3ZS05MjIzLTk3NDQxZTk4YmExMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwMjQ5MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDBiMDUxZWRiN2RhMTc1ZGU1MDQyYzVhYTgxODBjMWEwODkxNTcwODcxYWFiZDcyNmYyOWQ1ZDgyOTcxOGE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UkvMOf2bAjE4VmVLNh5qPQzPh-RGW4aZSi5hYLXYhl0) I am not getting +/- option

Please, read the guidelines. 
You are most likely not logged in as an owner account hence why you cannot see the +/- button. 
Feel free to reach out if you need further assistance.

luisarevalo21 commented:
> Hello
> Please ensure the frontend is running node 14.17.6.
> Also, follow the backend and run npm run build && npm run start to run and build the backend server.
> Reach out to me if you need further assistance.

Hello
Please ensure the frontend is running node 14.17.6.
Also, follow the backend and run npm run build && npm run start to run and build the backend server.
Reach out to me if you need further assistance.

luisarevalo21 committed:
resolved merge conflicts

samarth9201 reviewed (approved):
Verified the PR. LGTM!

![Screenshot (113)](https://github.com/user-attachments/assets/faa6b6cc-fc89-4c01-9319-903225f707fb)
![Screenshot (114)](https://github.com/user-attachments/assets/9f69d84f-d43e-470b-9073-abec245ddee8)
![Screenshot (115)](https://github.com/user-attachments/assets/12d75883-b56e-4c43-885a-eb59f11edc98)
![Screenshot (116)](https://github.com/user-attachments/assets/7a8aab2c-7621-49db-a4c1-1f8d598942f7)

sheetalmangate commented on src/components/Warnings/WarningsModal.jsx:
```diff
@@ -1,106 +1,106 @@
-/* eslint-disable react/no-unescaped-entities */
-/* eslint-disable react/destructuring-assignment */
-import { Modal, ModalHeader, ModalBody, ModalFooter, Button, Row } from 'reactstrap';
+// /* eslint-disable no-unused-vars */
```
Instead of commenting out all the code in the file, it would be better to delete the file.

Supriya1659 reviewed (approved):
Initially, I encountered an error, but after referring to the comments in the backend and executing npm run build and npm run start, I observed the warnings. I was also able to add and edit my own warnings, and everything is now working as expected.


![image](https://github.com/user-attachments/assets/b0a68d33-2e0e-49df-910f-1dfc0cfe201d)

![image](https://github.com/user-attachments/assets/d188c0e9-2239-4cff-9ccd-4acc16520c06)

![image](https://github.com/user-attachments/assets/b23f27ef-7945-49c4-b04c-5d289f87759c)

![image](https://github.com/user-attachments/assets/e95c5f69-1716-4d34-b6b1-6fd0300b7e90)

![image](https://github.com/user-attachments/assets/42aa1d61-1e3c-4f7f-9df0-29ffdd945640)

ajitp4292 reviewed (approved):
PR 2343 is working as expected. When I add a new warning tracker, I see that the tracking and warning tracker can be edited. Once edited, it displays the correct name.
<img width="1440" alt="Screenshot 2024-11-09 at 7 26 58 PM" src="https://github.com/user-attachments/assets/ef1c29c7-f491-4160-95e0-7e2f2282453b">
<img width="623" alt="Screenshot 2024-11-09 at 7 30 08 PM" src="https://github.com/user-attachments/assets/4e6c8ce7-6226-4dfc-947e-f9a4cf38d0f6">
<img width="927" alt="Screenshot 2024-11-09 at 7 30 32 PM" src="https://github.com/user-attachments/assets/6deba761-2e2f-49d4-ba8e-0eb0d4084c29">
<img width="371" alt="Screenshot 2024-11-09 at 7 30 51 PM" src="https://github.com/user-attachments/assets/06068e04-7d22-46aa-8bd2-45d1380b03fd">

Sheehan20 reviewed (commented):
Hi @luisarevalo21, great work on implementing the warning description management feature! Here's a detailed review:

1. Feature Highlights:
- Added functionality for owners to edit warning descriptions
- Implemented restrictions to prevent multiple simultaneous edits
- Created a new modal for warning tracker management

2. Key Improvements:
- Enhanced warning management workflow
- Added save mechanism for edited warnings
- Implemented owner-specific access controls

3. Technical Observations:
- Added new actions and reducers for warning description management
- Created modular components for warning interactions
- Implemented error handling for various warning-related operations

4. Potential Considerations:
- The PR mentions a bug with adding spaces to new warnings (to be addressed in PR #2037)
- Some reviewers noted occasional issues with warning creation and editing

5. Testing Recommendations:
- Verify warning creation across different scenarios
- Test edge cases for warning editing
- Ensure proper access control for warning management

Additional Notes:
- Recommended addressing the potential race condition in state management (as noted by @TXMO-dev)
- Consider adding more robust error handling

Overall, this is a solid implementation that improves the warning tracking functionality. Great job on the detailed implementation!

Srichand-Medagani reviewed (approved):
Works perfectly! I have left my comments and video on backend.

SSP24SCM16G reviewed (approved):
Everything is working good.
<img width="1470" alt="Screenshot 2025-03-01 at 19 47 35" src="https://github.com/user-attachments/assets/7c66fd28-1e8e-44de-8da5-85fb55c5df51" />
<img width="1470" alt="Screenshot 2025-03-01 at 19 47 50" src="https://github.com/user-attachments/assets/4f51a581-6ca7-47f1-afae-b2c2a4c0c8b4" />
<img width="1470" alt="Screenshot 2025-03-01 at 19 50 10" src="https://github.com/user-attachments/assets/82818aef-af01-4fa9-90ef-ccacc65e8b5a" />
<img width="1470" alt="Screenshot 2025-03-01 at 19 50 28" src="https://github.com/user-attachments/assets/be0a230e-498f-4e48-8757-e965dbc2dfbb" />

smohansatyaram reviewed (commented):
Owner edit permissions work as expected
Warning description updates correctly
Save functionality properly triggers backend updates

Note on Dark Mode Visibility:
In dark mode, the black text on dark backgrounds has low contrast, which may make it challenging for some users to read. (See last image)

![Screenshot 2025-03-27 192912](https://github.com/user-attachments/assets/e6f26bef-8464-426a-b749-0f29cbdfe679)
![Screenshot 2025-03-27 192927](https://github.com/user-attachments/assets/43aea490-f17e-4c4c-9243-323f74c2e8e9)
![Screenshot 2025-03-27 192941](https://github.com/user-attachments/assets/a6ecdcf2-980f-4858-a8a8-e7af7d69bfa3)
![Screenshot 2025-03-27 193035](https://github.com/user-attachments/assets/c4265f2f-08fb-49d7-a547-2c4f63207df3)

karnati007 reviewed (approved):
Tested the Warning Tracker feature by logging in as an owner account. Verified that clicking the "+/-" button correctly loads all current warnings. Successfully created a test warning, submitted it, and confirmed its appearance in the list. Edited the warning, saved changes, and verified that the updated warning reflected correctly in the user's warnings section and in the WarningTrackerModal.jsx. Cleared site data before testing. Everything is functioning as expected. 
![LuisPR2](https://github.com/user-attachments/assets/b5a5a495-ec82-4952-9722-e4b57256fa75)
![LuisPR](https://github.com/user-attachments/assets/88208725-f525-458b-ab71-4234eac0a12c)

sawantgayatri19 reviewed (changes requested):
Hi Luis,

Great work implementing the frontend support for adding, editing, and removing warning descriptions! I've reviewed the code changes and verified the workflow. 

Core Functionality:Logged in as an owner, clicking the `+/-` button correctly opens the tracking modal to add, edit, and save warning descriptions.
State & UI Updates: The transition from text elements to editable input fields within [WarningTrackerModal.jsx](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2343) handles updates cleanly.
Action Items Before Merge:
1. Merge Conflicts:Please resolve the active merge conflicts across the branch files (including `src/actions/warnings.js`, `src/components/Warnings/Warnings.jsx`, and `src/components/Warnings/modals/WarningTrackerModal.jsx`).
2. Async Dispatch Handling:Consider refactoring the nested dispatch calls in `src/actions/warnings.js` to mitigate potential race conditions, as flagged by earlier reviewers.
3. Code Cleanup: Ensure large blocks of commented-out code (e.g., in `WarningsModal.jsx`) are completely removed rather than left commented out.
4. Dark Mode Polish:Verify text contrast on warning modals under dark mode to ensure clear visibility.

Once these items and conflicts are cleared up, this will be ready to merge!
